src/server/dfly_main.cc

```diff
@@ -701,6 +702,24 @@ void PrintBasicUsageInfo() {
   std::cout << endl;
 }
 
+void ParseFlagsFromEnv() {
+  const auto& flags = absl::GetAllFlags();
+  for (const auto& flag_entry : flags) {
```

Comment:
for (const auto& [flag_name, flag] ...) saves you two lines 😎

Reply:
Also do:
```cpp
for (const auto& [flag_name, flag] : absl::GetAllFlags()) {
```
To save the above line :nerd_face:

Reply:
But... I would actually iterate [`environ`](https://man7.org/linux/man-pages/man7/environ.7.html) instead of flags, because:
1. It's much shorter (but it's not important as this is certainly not a hot path)
2. I think we should `exit()` in case a `FLAGS_X` exists as an environment variable, but the flag `X` does not exist, as this can hide a typo or unsupported feature

Reply:
+1 to keep stuff simple and scan the flags.

Reply:
> +1 to keep stuff simple and scan the flags.

I'd encourage you to reconsider the safety of iterating flags vs. iterating env-vars.
Think of the user running Dragonfly like this:
```bash
FLAGS_require_pass=... dragonfly ...
```

That user will get no feedback at all that they should have used `FLAGS_requirepass` (no underscore). Same goes for all typos of course.

Reply:
Shahar has a point, the idea of catching misnamed flags seems really useful. Though I'm not against keeping simplicity

Reply:
Why is iterating `environ` not simple? It's just a variable, is it not a matter of doing:

```cpp
for (char **env = environ; *env != nullptr; env++) {
  ;;;
}
```
?